### ISSUE TITLE
allow path list instead of glob pattern as `source` for `pl.scan_parquet`

### ISSUE THREAD

**Issue by sslivkoff** [pola-rs/polars#8047]
### Problem description

multi-file parquet scans currently require use of glob patterns. but globs are a bit limited in the types of selections they can make, especially since python's `glob.glob()` cannot use curly brace expressions like posix-style globs

if `scan_parquet()` could accept a list of parquet paths instead of a single glob pattern, this would have significant performance benefits for a couple use cases:
1. you are scanning a large directory of parquet files, and you already know the specific subset of files that contain all of the relevant rows for the scan. e.g. you have a directly of 2000 parquet files, but only 10 are relevant. even if your parquet files are indexed with statistics, it can still take a long time to scan through 2000 files instead of just picking the relevant ones from the start
2. your dataset is partitioned across multiple directories due to operational constraints. you can always scan each directory separately and then manually concat. but this is less elegant than polars glob scans that do the concating transparently and with builtin parallelism

with current polars, possible workarounds include:
1. just use an overly-inclusive blob. this can be slow if there is a lot of irrelevant extra data but works for all types of query operations
2. perform multiple scans and manually concat. this can give >10x speedup depending on your situation. but requires extra steps and isn't suitable for all types of query operations. this can also break or complicate streaming
3. create a tempdir filled with symlinks to all of the parquet files of interest, and then just use a glob to select everything in the tempdir. this works for many cases and maintains the ability to stream larger-than-memory queries. but feels a bit hacky and only works for local files

**Comment by cjackal** [comment]
If you have the paths of interest in advance, you can simply take scan_parquet to each path then collect_all(or concat and collect, depending on your goal).

e.g)
```python
paths: list[str]  # whatever list of paths
df_all = pl.concat([pl.scan_parquet(path) for path in paths]).collect()
```

**Comment by kcajf** [comment]
This seems fixed in 1.0.0 - `scan_parquet` now accepts a list of paths.

**Comment by sslivkoff** [comment]
yep. thanks for pointing that out
